HCCGo/app/js/jobViewCtrl.js

```diff
@@ -13,17 +33,22 @@ jobViewModule.controller('jobViewCtrl', ['$scope', '$log', '$timeout', 'connecti
       }
       else if (result.hasOwnProperty("outText") && result.hasOwnProperty("errText")) {
         $scope.job = result;
+        $scope.outDownload = true;
+        console.log($scope.outDownload);
       }
       else {
         // In parallel, get the size of the output and error
         connectionService.getFileSize(result.outputPath).then(function(size) {
           // If the file is larger than 5MB
           if(size > 5*1025*1024) {
-            result.outText = "The Output file is too large to be displayed here."
+            result.outText = "The Output file is too large to be displayed here.";
+            $scope.outDownload = false;
           } else {
+
             connectionService.getFileText(result.outputPath).then(function(data) {
               var text = data.length>0 ? data : "(none)";
               result.outText = text;
+              $scope.outDownload = true;
               db.update(
```

Comment:
I think you should set the outDownloaded flag in the DB.   then you wouldn't have to check it next time the person loads the job information.  Same with error.